contracts/settling_game/ModuleController.cairo

```diff
@@ -61,85 +63,121 @@ func external_contract_table(external_contract_id: felt) -> (address: felt) {
+// @return proxy_admin: Proxy admin address
+@external
+func initializer{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+    arbiter_address: felt, proxy_admin: felt
 ) {
     arbiter.write(arbiter_address);
 
     // set genesis
     let (block_timestamp) = get_block_timestamp();
     genesis.write(block_timestamp);
 
-    // write patterns known at deployment. E.g., 1->2, 1->3, 5->6.
-
-    // resources logic to settling state
-    can_write_to.write(ModuleIds.Resources, ModuleIds.Settling, TRUE);
-
-    // combat can write to resources
-    can_write_to.write(ModuleIds.L06_Combat, ModuleIds.Resources, TRUE);
-
-    // # combat can write to settling
-    can_write_to.write(ModuleIds.L06_Combat, ModuleIds.Settling, TRUE);
-
-    // settling can write to s realms tokens
-    can_write_to.write(ModuleIds.Settling, ModuleIds.S_Realms_Token, TRUE);
-
-    // resources can write to resources token
-    can_write_to.write(ModuleIds.Resources, ModuleIds.Resources_Token, TRUE);
-
-    // buildings can write to resources token
-    can_write_to.write(ModuleIds.Buildings, ModuleIds.Resources_Token, TRUE);
-
-    // combat can write to resources token
-    can_write_to.write(ModuleIds.L06_Combat, ModuleIds.Resources_Token, TRUE);
-
-    // crypts can write to crypts token
-    can_write_to.write(ModuleIds.L07_Crypts, ModuleIds.S_Crypts_Token, TRUE);
-
-    // crypts resources can write to resources token
-    can_write_to.write(ModuleIds.L08_Crypts_Resources, ModuleIds.Resources_Token, TRUE);
-
-    // # crypts logic to resources
-    // can_write_to.write(ModuleIds.L07_Crypts, ModuleIds.L08_Crypts_Resources, TRUE)
-
-    // # resources logic to crypts state
-    // can_write_to.write(ModuleIds.L08_Crypts_Resources, ModuleIds.L07_Crypts, TRUE)
-
-    // Lookup table for NON module contracts
-    external_contract_table.write(ExternalContractIds.Lords, _lords_address);
-    external_contract_table.write(ExternalContractIds.Realms, _realms_address);
-    external_contract_table.write(ExternalContractIds.S_Realms, _s_realms_address);
-    // external_contract_table.write(ExternalContractIds.Crypts, _crypts_address)
-    // external_contract_table.write(ExternalContractIds.S_Crypts, _s_crypts_address)
-    external_contract_table.write(ExternalContractIds.Resources, _resources_address);
-    external_contract_table.write(ExternalContractIds.Treasury, _treasury_address);
+    Proxy.initializer(proxy_admin);
+    return ();
+}
 
+// @notice Set new proxy implementation
+// @dev Can only be set by the arbiter
+// @param new_implementation: New implementation contract address
+@external
+func upgrade{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+    new_implementation: felt
+) {
+    Proxy.assert_only_admin();
+    Proxy._set_implementation_hash(new_implementation);
     return ();
 }
 
-//###########
-// EXTERNAL #
-//###########
+// // -----------------------------------
```

Comment:
Since it's all commented, can it be removed altogether?

Reply:
yes good point, was missed